tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1179,6 +1180,46 @@ class ResizeMethod(object):
   MITCHELLCUBIC = 'mitchellcubic'
 
 
+def _check_equal_input_output_dtypes(func):
+  """This decorator issue will issue a warning if the input dtype and output
+  dtype are different. Help to prevent silent dtype conversion from integer to
+  floating point format."""
+
+  def wrapper(*args, **kwargs):
```

Comment:
A decorator is very ugly in stack traces, can we make this into a utility function which is called from `_resize_images_common` instead?

Reply:
Not easily doable. We need to access the input and output dtypes to compare them.
Might not be ideal but this is really a case for decorators... Aside of this I would prefer not touching the logic of these functions.
If you really don't want a decorator, can you be more specific on the solution you have in mind ?